GVRf/Extensions/3DCursor/IODevices/gvrmobileiodevice/src/main/java/com/example/gvrmobileiodevice/GVRMobileIoDevice.java

```diff
@@ -0,0 +1,193 @@
+package com.example.gvrmobileiodevice;
+
+import android.content.Context;
+import android.content.IntentFilter;
+import android.hardware.Sensor;
+import android.hardware.SensorManager;
+import android.net.wifi.p2p.WifiP2pManager;
+import android.net.wifi.p2p.WifiP2pManager.ActionListener;
+
+import org.gearvrf.GVRContext;
+import org.gearvrf.io.cursor3d.CursorEvent;
+import org.gearvrf.io.cursor3d.CursorEventListener;
+import org.gearvrf.io.cursor3d.IoDevice;
+import org.gearvrf.io.cursor3d.OutputEvent;
+import org.gearvrf.utility.Log;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+/**
+ * Created by v.nidamarthy, d.knaan, p.mehta, r.rudhradevan on 8/26/16.
+ *
+ * Server - mobile with app, client - mobile/iodevice
+ * This class encodes the reading and writing of data to the app over Wifi,  //TODO adding bluetooth communication
+ * WifiDirectBroadcastReceiver establishes connection over WiFi
+ * Connect Server listens to the client and establishes connection and
+ * also the Input and Outputstreams
+ * ClientWrite writes data/feedback to the client
+ * ReadFromServer reads the incoming data i.e the orientation
+ *
+ */
+public class GVRMobileIoDevice extends IoDevice{
+    public static final String DEVICE_ID = "gvrmobile";
+    public static final int VENDOR_ID = 525;
+    public static final int PRODUCT_ID = 0;
+    public static final String VENDOR_NAME = "Samsung";
+    public static final String DEVICE_NAME = "GVR Mobile Device";
+    private static final String TAG = GVRMobileIoDevice.class.getSimpleName();
+    private final ServerInfo serverInfo;
+    private Context androidContext;
+
+    /* Fields for sensors and sockets */
+    private SensorManager mSensorManager;
+    private Sensor accelerometer;
+    private Sensor magnetometer;
+    private GVRContext mGVRContext;
+
+    /* Broadcast receiver side*/
+    private WifiP2pManager mManager;
+    private WifiP2pManager.Channel mChannel;
+    private IntentFilter mIntentFilter;
+    public final static int port = 8888;
+    private WifiDirectBroadcastReceiver mReceiver;
+
+    public enum Actions {ACTION_VIBRATE}   //enum to set feedback actions to the device
+
+    @Override
+    public void handleOutputEvent(OutputEvent event) {
+
+        try {
+            if (event.action == Actions.ACTION_VIBRATE.ordinal())
```

Comment:
You might want to pass the ACTION_VIBRATE ordinal number to the ClientWrite class so that it uses the actual action value rather than a hardcoded value.